src/persist-types/src/columnar.rs

```diff
@@ -305,26 +305,31 @@ pub trait ColumnDecoder<T> {
 pub trait ColumnEncoder<T> {
     /// Type of column that this encoder returns when finalized.
     type FinishedColumn: arrow::array::Array + Debug + 'static;
+    /// Type of statistics this encoder returns when finalized.
+    type FinishedStats: DynStats + 'static;
```

Comment:
All callers of `finish` immediately convert the result of this to `ColumnarStats`. (Which is great - love that new type.) Could we make `finish` return `ColumnarStats` directly and skip the intermediate / associated type?

Reply:
Good thought! I tried to do this originally but what throws a small wrench in it is `OptionStats`. It's nice to have the implementation for `OptionStats` require the the inner type `impl Into<ColumnStatsKind>` to prevent nested `OptionStats`. Not an impossible problem to solve but a little tricky